Infrastructure: update dofile(), loadfile(), os.execute() handling of path/filenames with special characters

SlySven opened the pull request:
This will close #5705.

The update also fixes some issues with the Lua functions (for Lua 5.1):
* os.execute(...)
* dofile(...)
* loadfile(...)
in the same manner that the ***upstream*** author and @vadi2 already did for io.lines(...) in #2756, see also: https://gist.github.com/Egor-Skriptunoff/2458547aa3b9210a8b5f686ac08ecbf0#gistcomment-2969093

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

#### Release post highlight
Infrastructure update handling of accented path/filenames for three standard Lua library functions that access the OS 's filesystem.

*Edited to reflect that it was the upstream "author" and not me!*

SlySven committed:
Infrastructure: (Public) update utf8_filenames.lua

This will close #5705.

The update also fixes some issues with the Lua functions (for Lua 5.1):
* os.execute(...)
* dofile(...)
* loadfile(...)
in the same manner that the author and @vadi2 already did for io.lines(...)
in #2756, see also:
https://gist.github.com/Egor-Skriptunoff/2458547aa3b9210a8b5f686ac08ecbf0#gistcomment-2969093

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented on src/mudlet-lua/lua/utf8_filenames.lua:
```diff
@@ -300,61 +303,6 @@ local function modify_lua_functions(all_compressed_mappings)
             return orig_io_output(file)
          end
 
-         local orig_lfs_attributes = lfs.attributes
```
Why is all of this bring removed?

SlySven replied:
I don't know - that is just what the upstream author has done. Maybe it doesn't work? 🤷‍♂️

vadi2 replied:
No it was actually I who added them as part of https://github.com/Mudlet/Mudlet/pull/2709. Let's document this better so we don't remove it by accident, and of course put them back.

SlySven replied:
Okay - I've put it back... 😀 

Maybe we should offer it back to the upstream author...

SlySven replied:
Unfortunately the commit was not signed (not showing "Verified") as I am out and about and I forgot to grab my OpenGPG card to allow me to sign it... 🙄

SlySven replied:
Will upstream accept them - that would be the best outcome.

SlySven committed:
Partial revert: restore our additions to the file

We do not use the upstream code as given - we have added some stuff
of our own which needs to be kept.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented on src/mudlet-lua/lua/utf8_filenames.lua:
```diff
@@ -164,9 +165,17 @@ local function modify_lua_functions(all_compressed_mappings)
-   if getOS() == "windows" then
+   if (os.getenv"os" or ""):match"^Windows" then
 
-      local codepage = getWindowsCodepage()
```
I would also keep this change because it uses a built-in Mudlet function, instead of opening a `cmd` window which is annoying for the user.

After this change is reverted... what exactly is this improvement bringing?

SlySven replied:
Well it is accessing the appropriate key in the registry which is the *right-thing* on Windows for getting settings... 

but looking into our function I see it is doing the same thing so ours is better - but it does mean that we have forked from upstream. **However we are still using elements of that which does not have a GPL-like license so we still need to include the license for the original work...**

SlySven committed:
Fix: correct a member name from mpSpltter... to mpSplitter...

Also remove a pair of `Q_UNUSED()` macros as the arguments they work on
*are* used in a debugging message.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
Bother, I forgot to save the edited commit message before I make the last commit and thus used the previous one - so it is totally unconnected, the message should have been:
> Update: remove reversion of our own customisations
>
> Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

so I will have to take care to properly clean up the commit message if/when it is *squashed-and-merged*.

vadi2 committed:
Merge branch 'development' into Infrastructure_Public_update_utf8_filenames.lua

vadi2 reviewed (approved):
Let's give it a go.
